core/src/main/java/org/elasticsearch/tasks/Task.java

```diff
@@ -132,4 +135,16 @@ public Status getStatus() {
     }
 
     public interface Status extends ToXContent, NamedWriteable {}
+
+    public TaskResult result(DiscoveryNode node, Throwable error) throws IOException {
+        return new TaskResult(taskInfo(node, false), error);
```

Comment:
Why not `true`? I think it'd be nice to have the detailed info. Actually I pretty much always want it all the time.